File: src/vlab/server_js/check/PreCheckProcessorImpl.js
function  preCheck(conditions, instructions, generatingResult){
    var preCheckResult = Java.type('rlcp.server.processor.check.PreCheckProcessor.PreCheckResult');

    conditions.forEach(function(conditionForChecking){
        print('id: ' + conditionForChecking['id']);
        print('time: ' + conditionForChecking['time']);
        print('input: ' + conditionForChecking['input']);
        print('output: ' + conditionForChecking['output']);
    });

    print(instructions);

    print(generatingResult['text']);
    print(generatingResult['code']);
    print(generatingResult['instructions']);

    var result = new preCheckResult("preCheckResult");
    return result;
}

File: src/vlab/server_java/check/PreCheckProcessorImpl.java
package vlab.server_java.check;

import rlcp.check.ConditionForChecking;
import rlcp.generate.GeneratingResult;
import rlcp.server.processor.check.PreCheckProcessor;

import java.util.List;

/**
 * Simple PreCheckProcessor implementation. Supposed to be changed as needed to provide
 * necessary before checking support.
 */
public class PreCheckProcessorImpl implements PreCheckProcessor<String> {
    @Override
    public PreCheckResult preCheck(List<ConditionForChecking> conditions, String instructions, GeneratingResult generatingResult) {
        return new PreCheckResult("");
    }
}
